### ISSUE TITLE
Windows 10 Support

### ISSUE THREAD

**Issue by dave-tucker** [docker/toolbox#77]
### Download a test version of the Toolbox for Windows 10 [here](https://github.com/docker/toolbox/releases/download/v1.8.1a/DockerToolbox-1.8.1a-windows-10-test.exe).

Currently, the latest stable version of VirtualBox that is shipped in the toolbox does not support Windows 10.

A user trying to use Toolbox on Windows 10 will be able to install the package succesfully, but the VM will not be able to be created because of the following issue: [virtualbox#14040](https://www.virtualbox.org/ticket/14040)

This appears to be fixed in the latest test build `5.0.1r102010`

Suggestions:
- [x] Disable installation of Toolbox on Windows 10 for now
- [x] Release an experimental version of toolbox that packages the latest test build of virtualbox
- [x] Remove any installation restrictions when a stable version of VirtualBox ships that supports Windows 10

**Comment by dave-tucker** [comment]
As an FYI, anybody who wants to manually make this work on Windows 10 can:

1) Download and install the latest test build of Virtualbox [here](https://www.virtualbox.org/wiki/Testbuilds)
2) `docker-machine rm default`
3) Open `Kitematic` and let it create a new VM
4) Test that it's working - I deployed a Jenkins container and it seems to be ok!

**Comment by lexxpavlov** [comment]
I have win10. I installed latest Virtualbox (5.0.1r102010), then try to run Kitematic. Kitematic first removes this my Virtualbox and installs 5.0.0r101573 instead. If I setup Virtualbox via Kitematic, then I have several troubles with it. I can't run Kitematic with new Virtualbox (r102010) without reinstall to previous Virtualbox.

**Comment by jeffdm** [comment]
@lexxpavlov thanks for the note, looks like there may be a bug in how KM checks VBox versions.

**Comment by StefanScherer** [comment]
@jeffdm VirtualBox 5.0.2 is available. I've just seen a new Chocolatey package.

**Comment by lexxpavlov** [comment]
@StefanScherer I insalled VirtualBox 5.0.2 (not test build), but Kitematic don't want use it. 
I created a issue in Kitematic repo (https://github.com/kitematic/kitematic/issues/932).

**Comment by iprastha** [comment]
Just confirmed this on Windows 10, successfully run my container using dave's comment.
However in my case, using "docker-machine rm default" was not successful (seems like stuck forever), so I fired up virtualbox user interface and removed the default VM from there, then run kinematic which created a working VM.

**Comment by Grafikart** [comment]
I think the warning shouldn't prevent the installation, if we want to install only docker-machine to work with Hyper-V on Windows 10.

**Comment by mysticmind** [comment]
Docker team have released a [Windows 10 test version](https://github.com/docker/toolbox/releases/download/v1.8.1a/DockerToolbox-1.8.1a-windows-10-test.exe) which already uses Virtualbox (5.0.1r102010). If you are installing fresh, you could directly download and use this version.

Everything works fine using the above installer except for Kitematic which on launch messes up the host VM. I would suggest not to use Kitematic for the time being.

**Comment by lexxpavlov** [comment]
@mysticmind This version was released yesterday, I tried it. It version has VirtualBox 5.0.0r101573, as I said above.

**Comment by dgmdevelopment** [comment]
I have the Win 10 test version working great. if you have any version of the toolbox installed prior to the 1.8.1a - uninstall it first. Then install VBox 5.0.1r102010.  Remove the old default vm if present.  I already had Mysys-git on the machine. Install the Win 10 test version, uncheck Vbox and Mysys-git naturally because already installed. Both the Quickstart and Kitematic fire up. Hope that works for you too!

**Comment by luisrudge** [comment]
I'm having this error on a windows 10 box:

![image](https://cloud.githubusercontent.com/assets/941075/9296721/63601b64-4469-11e5-9920-0b78c5f84b0a.png)

**Comment by luisrudge** [comment]
With the 1.8.1b release, it gets stuck here:

![image](https://cloud.githubusercontent.com/assets/941075/9296804/085550f2-446b-11e5-9451-9b66d8c6818d.png)

![image](https://cloud.githubusercontent.com/assets/941075/9296806/11e47058-446b-11e5-9e4b-6bd5019861e4.png)

**Comment by luisrudge** [comment]
If I recreate the VM, I get the same result as https://github.com/docker/toolbox/issues/77#issuecomment-131652911

**Comment by luisrudge** [comment]
So, I just formatted my pc. Installed Windows 10 from scratch and docker toolbox 1.8.1b. Everything works as expected! I'm not sure what was causing the issue, but I can confirm this release works on win10.

**Comment by luisrudge** [comment]
Sadly, it stops working after I install visual studio :(

**Comment by luisrudge** [comment]
Sorry to comment again. I removed the virtualbox machine and opened kitematic again and it's working again :)

**Comment by bpuglisi** [comment]
Wanted to share my solution that worked when all other Windows 10 ones out there weren't.   Docker Toolbox + Docker Compose - Enjoy!

http://brianpuglisi.com/how-to-get-docker-compose-working-on-windows-10/

**Comment by StefanScherer** [comment]
@bpuglisi Thanks for sharing! So is it possible to build a docker-compose.exe with pyInstaller with your setup?

**Comment by jeffdm** [comment]
Closing for now since there's a version of VirtualBox that works with Windows 10

**Comment by gbisheimer** [comment]
Hi. I'm still having problems creating host-only interface using DockerToolbox-1.8.1b running on Windows 10.0.10240. I also tried using VirtualBox testbuild 5.0.3-102322 without luck. I followed every troubleshooting guide I found on the internet and couldn't make it work.

Neither it works creating the interface directly from VirtualBox interface. The interface actually gets created on Windows, but VirtualBox is not aware of that, as it isn't listed on the host-only interface list.

I also have VMWare installed on the same machine. There are any known incompatibilities with this software?

Any help will be appreciated!!

Here is the console transcript after executing `Docker Quickstart Terminal`:
```
STDERR:
Creating VirtualBox VM...
Creating SSH key...
Creating disk image...
Creating 20000 MB hard disk image...
Converting from raw image file="stdin" to file="C:\Users\gbisheimer\.docker\machine\machines\default\disk.vmdk"...
Creating dynamic image with size 20971520000 bytes (20000MB)...
executing: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe createvm --basefolder C:\Users\gbisheimer\.docker\machine\machines\default --name default --register
STDOUT: Virtual machine 'default' is created and registered.
UUID: e8838696-b567-4f58-9ba8-85a7705db7cc
Settings file: 'C:\Users\gbisheimer\.docker\machine\machines\default\default\default.vbox'

STDERR:
VM CPUS: 1
VM Memory: 2048
executing: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe modifyvm default --firmware bios --bioslogofadein off --bioslogofadeout off --bioslogodisplaytime 0 --biosbootmenu disabled --ostype Linux26_64 --cpus 1 --memory 2048 --acpi on --ioapic on --rtcuseutc on --natdnshostresolver1 off --natdnsproxy1 off --cpuhotplug off --pae on --hpet on --hwvirtex on --nestedpaging on --largepages on --vtxvpid on --accelerate3d off --boot1 dvd
STDOUT:
STDERR:
executing: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe modifyvm default --nic1 nat --nictype1 82540EM --cableconnected1 on
STDOUT:
STDERR:
using 192.168.99.1 for dhcp address
executing: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe list hostonlyifs
STDOUT:
STDERR:
executing: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe hostonlyif create
STDOUT:
STDERR: 0%...
Progress state: E_FAIL
VBoxManage.exe: error: Failed to create the host-only adapter
VBoxManage.exe: error: Code E_FAIL (0x80004005) - Unspecified error (extended info not available)
VBoxManage.exe: error: Context: "enum RTEXITCODE __cdecl handleCreate(struct HandlerArg *)" at line 71 of file VBoxManageHostonly.cpp

Error creating machine: exit status 1
You will want to check the provider to make sure the machine and associated resources were properly removed.
Looks like something went wrong... Press any key to continue...
```

**Comment by visumagic** [comment]
After lot of trial and errors I am able to run Docker (Docker Toolbox , Kitamatic  ) in Windows 10.

Solution:
VirtualBox-4.3.30-101610-Win.exe

Remove your old virtualbox and install the above version. 

thank you

**Comment by wiradikusuma** [comment]
I installed VirtualBox-5.0.3-102467-Win.exe and DockerToolbox-1.8.1c.exe (deselecting Virtual Box). It doesn't work.

Creating Machine default...
Creating VirtualBox VM...
Creating SSH key...
Starting VirtualBox VM...
Starting VM... <-- stays there forever

Anyone?

**Comment by phauer** [comment]
I have the same problem with VirtualBox-5.0.3-102467-Win.exe as wiradikusuma. Using VirtualBox-4.3.30-101610-Win.exe as visummagic supposed doesn't work either.

**Comment by javafun** [comment]
Same as phauer. Neither 5.0 or latest 4.3 worked.

**Comment by wiradikusuma** [comment]
I installed the Toolbox yesterday on Windows 7 (I still have problem with Windows 10 as well), same problem.

**Comment by visumagic** [comment]
First time, it worked for me. I was able to see nice UI of Toolbox. But not
again same issue. I was able to install containers from kitematic UI. After
couple of days, it's not working. Now I switched to my old technique, using
Virtualbox with Ubuntu inside it which has docker.

=RrAaGgHhUu

**Comment by dheerajbhaskar** [comment]
I followed @dave-tucker 's method. When I click on docker quickstart terminal shortcut, it gets stuck at the point mentioned by @wiradikusuma . If I remove the vm from virtualbox and start kitematic, it tries to download its own version of virtualbox, then I get the following error:

    C:\Program Files\Docker Toolbox\kitematic\resources\resources\docker-machine.exe -D create -d virtualbox --virtualbox-memory 2048 default returned non zero exit code. Stderr: Error creating machine: exec: "VBoxManage": executable file not found in %PATH% You will want to check the provider to make sure the machine and associated resources were properly removed.

![kitematic error](https://cloud.githubusercontent.com/assets/2944909/9766487/c6a9a3d6-5736-11e5-8220-0071a77bfc39.png)

Has anyone figured this out?

**Comment by dheerajbhaskar** [comment]
- I've uninstalled everything
- Restarted
- installed Vbox 5.0.4
- Restarted
- installed docker toolbox 1.8.1(c)
- Launched quickstart terminal shortcut
- stuck on starting vm; the vm says its running in the vbox interface though
- disappointed

I see a few people have gotten it to work on win 10 in this thread and elsewhere on the internets. How is that?

**Comment by yourchanges** [comment]
docker toolbox 1.8.2
Starting VM... <-- stays there forever

**Comment by ghal** [comment]
I have also the same problem...

**Comment by dheerajbhaskar** [comment]
I went back to using boot2docker with 5.0.2 vbox. Seems to be working so
far.

**Comment by rijkaardorisme** [comment]
I have managed it to work on my Windows 10 Pro 64bit
I've used a test build found on a post from the [Docker Community Forum](https://forums.docker.com/t/docker-toolbox-on-windows-10-is-it-possible/2822)

**Comment by dheerajbhaskar** [comment]
Hi Rijkaard, could you please share the steps which got your setup working?

**Comment by SkiftCreative** [comment]
I am still having this issue.l just downloaded toolbox and installed the latest test build of Virtualbox and installed the guest additions addon as well. 

@jeffdm  So I do not belive this issue should be marked as closed. I have also tried with the normal VBox release as well and get the same results. Other people seem to be having this issue as well.

**Comment by painhardcore** [comment]
DockerToolbox-1.8.2a not working on windows 10 and windows 8.1
![docker](https://cloud.githubusercontent.com/assets/4718161/10031625/10fbb758-6187-11e5-97d9-d1a4f09ee707.png)

**Comment by painhardcore** [comment]
Some strange errors.
Second when i remove VM and trying to start kinematic as administrator
![docker2](https://cloud.githubusercontent.com/assets/4718161/10031933/968f487a-6188-11e5-97db-311b710bf729.png)
![docker3](https://cloud.githubusercontent.com/assets/4718161/10031934/968f53c4-6188-11e5-9390-8749896e578d.png)

**Comment by hakabe** [comment]
Workaround is to install the Docker Toolbox normally and then uninstall Virtualbox, plus the %userprofile%/.Virtualbox folder. Download the newest Virtualbox testbuild from https://www.virtualbox.org/wiki/Testbuilds (5.x) and install it.

Kinematic gave me error on the first run. Just click Delete vm and retry setup. There, you are set.

**Comment by git-bub** [comment]
I just ran into the issue others are reporting of getting stuck on the "Starting VM..." step when trying to launch the Docker Toolbox (Docker Quickstart Terminal) and was able to solve my situation.  Your situation may be different, but for me the issue was that virtualization was disabled on my CPU.

I found this by opening VirtualBox while docker was stuck on that Starting VM step, and saw that the VM (named default) was running so I clicked the Show button in the toolbar.  When it attempted to show the VM I got an error message from VirtualBox with the details regarding the virtualization issue and so I checked my BIOS and found I needed to enable it.  Potentially you could go the opposite route and disable hardware virtualization for the docker VM and try launching the Docker Toolbox again, but I have not tested this.

Even if your issue is not virtualization, you may be able to get some details about what is holding you up by going into VirtualBox while stuck on that Starting VM step.  Good luck.

**Comment by jouana** [comment]
Hi,
For start sorry for my english isn't my best skill.
So, I made what dave said on the top of this topic but before launch kitematic i deleted all virtualbox host-only eternet adapter and in control panel -> programe & features -> windows features, disabled Hyper-V

hope this can help you
+

**Comment by extensionsapp** [comment]
Windows 10
VirtualBox-5.0.5-103003-Win
DockerToolbox-1.8.2c

```
Machine default already exists in VirtualBox.
Starting machine default...
Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
exit status 255

Looks like something went wrong... Press any key to continue...
```

**Comment by daodao10** [comment]
Please make sure intel virtualization technology enabled in bios.
http://www.sysprobs.com/disable-enable-virtualization-technology-bios

**Comment by whatwho** [comment]
I've tried to install docker toolbox on a windows 10 machine running in VirtualBox on OS X, but had the same problems as above. The interesting thing is, that I also tried to install it on my wife's windows 7 machine, and had no luck, the problem is the same (creating vm halts, kitematic also at 99%).

**Comment by jouana** [comment]
Hi, delete the vm and relaunch.
Have a good day.
Antoine jouan

**Comment by danistefanovic** [comment]
I've had the same problem with the stucking `Starting VM...`.  When I opened the `default` VM in VirtualBox, I've seen that the boot2docker whale banner was printed. It seems that it somehow worked but the Docker Quickstart Terminal was not "notified" about that.

The following steps fixed it for me:
1. Install the newest [VirtualBox test build](https://www.virtualbox.org/wiki/Testbuilds)
2. Delete the `default` machine: `docker-machine rm default`
3. Start `Kitematic` and let it create the `default` virtual machine again

Don't forget to connect your shell with the `default` machine: `eval $(docker-machine env default)`

**Comment by cmlval** [comment]
Hello, i have a query about docker on windows, recently i installed it but i want to change the folder .docker to another place for example D:/.docker ... is posible to do that???

**Comment by Kapusch** [comment]
@danistefanovic  It worked for me, thanks a lot ! Too much time spoiled for that, it was great to read your answer.

**Comment by numediaweb** [comment]
@danistefanovic Thanks! Installing latset  virtualbox version fixed it; VirtualBox-5.0.9-103625-Win.exe on windows 10 (64bit).

**Comment by greha** [comment]
@danistefanovic 
How can i run the eval $(docker-machine env default) command on a win10 box?

**Comment by danistefanovic** [comment]
@greha Use Git Bash/MinGW

**Comment by jjmartin** [comment]
i'm finding that after starting kitmatic and havingit build the vm and get docker running - docker-machine does not show the vm running when in administrator powershell but non-admin shows it fine

**Comment by jjmartin** [comment]
@greha  if you run from powershell - you can do `docker-machine env --shell powerhsell default |Invoke-expression`

**Comment by kjitin** [comment]
Got docker 1.9.0c working on windows 10, first installed docker ...removed the virtualbox 5.0.1 associated with it...uninstalled it from the machine....restarted windows...and installed VirtualBox 5.0.11-104101....docker started working after it detected the new Virtualbox

**Comment by francoishill** [comment]
See if there any processes (images) with the name `docker-machine-driver-virtualbox.exe` running in Windows' Resource Monitor under the Network section. Mine got stuck at "busy creating" for about 30 minutes while downloading this.

**Comment by nitind** [comment]
@greha Append "--shell cmd", e.g. docker-machine env default --shell cmd, will emit something that works there.

**Comment by dreamcat4** [comment]
Installed via chocolatey pkg manager:

* VirtualBox 5.10
* docker client
* docker compose
* docker machine

Installed docker toolbox silently with args: `/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP-`

Then:

![start.sh](https://i.imgur.com/scwjpO3.png)

and

![KiteMatic](https://i.imgur.com/Wl53lnU.png)

I think I'll try again after reinstalling Windows 10. But AFAIK the error message from Kiematic isn't right. As docker-machine.exe files are all present in `C:\Program Files\Docker Toolbox` folder.

**Comment by jjmartin** [comment]
still having issues with win 10 now - i don't know if recent updates screwed it up or what - but it was working at one point, now i have uninstalled and resinstalleed toolbox and virtualbox a number of times and can't seem to get it to go... i also seemed to run into this issue on virtual box - https://forums.virtualbox.org/viewtopic.php?f=6&t=70199   
but i don't know if anything i did on that actually got me any closer to running docker again...

**Comment by Zugwalt** [comment]
Latest docker toolbox  for windows I got, version 1.10.1 came with VirtualBox 5.0.14. Didn't install Kitematic (I used cli) and git (already had it installed). Worked out of the box for me.

**Comment by ryanmaclean** [comment]
This appears to have reared it's head again: Windows 10, no virtualbox installed, clean install, virtualization enabled in BIOS:

```
Running pre-create checks...
Error with pre-create check: "Hyper-V is installed. VirtualBox won't boot a 64bits VM when Hyper-V is activated. If it's installed but deactivated, you can use --virtualbox-no-vtx-check to try anyways"
Looks like something went wrong in step ´Checking if machine default exists´... Press any key to continue...
```

**Comment by SkiftCreative** [comment]
I have the same issue as @ryanmaclean . I have a clean install of windows 10 with virtualization enabled.  I also get it when dual booting with bootcamp on a 2013 macbook pro.

**Comment by JohnBernardsson** [comment]
@ryanmaclean  @SkiftCreative For me this was fixed disabling Hyper-V: http://www.eightforums.com/tutorials/42041-hyper-v-enable-disable-windows-8-a.html

I had to disable it anyways to use Android emulator with x86 emulation so...

Hope this helps!

**Comment by azarudeen90** [comment]
@JohnBernardsson Thanks for link with clear cut idea to disable Hyper-V setting completely.

 It clearly solved my problem in my Windows 10 system. But still its just a hard way as the change setting needs a sys restart. My VMware workstation does needs the Hyper-V for startup certain machines.

**Comment by scetix** [comment]
Solution for me was to discover that **HV Host Service** was not running/would not start. After googling i found this:

http://www.rovermarinho.org/hyper-v-could-not-be-started-because-the-hypervisor-is-not-running/

TLDR;
Open CMD in Admin mode, run `BCDEDIT /set hypervisorlaunchtype auto` and restart

**Comment by ruthyaneth** [comment]
Install docker, tensorflow and python 
https://youtu.be/OiPjSnKAmSI

**Comment by ruthyaneth** [comment]
https://youtu.be/OiPjSnKAmSI
